### DISCUSSION TITLE
Get points from a curve with exact values

### DISCUSSION THREAD

**Discussion by valentinavalenzi** [jsvine/pdfplumber#1079] [Ask for help with specific PDFs]
Hi. I am trying to use pdfplumber in order to take curves from a pdf and get its points in order to use them later. The problem is that pdfplumber gives me these points but with the pdf coordinate (using page.curves I get the coordinates of the curve). 
I would like to know how can I get the points with the corresponding axes of the graph. 
For example, here the highest point would be (165;650) but  using pdfplumber I get the coordinates inside the pdf. How can I make something that gets the values I am looking for?
![image](https://github.com/jsvine/pdfplumber/assets/92260481/b583087b-9f37-4e86-ae32-13d9bc26b8dd)

**Comment by jsvine** [comment]
Hi @valentinavalenzi, and thanks for your interest in `pdfplumber`. This is an interesting challenge, because PDFs don't have any concept of a chart — here, the chart is just a combination of text, lines/curves, et cetera. Using `page.lines` (I'm guessing, but cannot confirm without access to the PDF itself), you may be able to identify the position of the horizontal gridlines; you could use the position of those to deduce the values of the other points in the chart.

**Reply by valentinavalenzi** [reply]
Yes I was thinking something about that. Maybe that way I will have an approximation. Thank you very much for your help!

**Reply by petermr** [reply]
I have worked on exactly this problem and depending on the image can extract axes and points. There are heuristics involved and there is a slight problem with curves as we cannot extract control points (I think this is a `pdfminer` problem - may have been fixed). If you have a lot of graphs like this - from the same source - it may be worth collaborating
